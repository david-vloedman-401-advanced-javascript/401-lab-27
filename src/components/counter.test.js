import React from 'react';
import renderer from 'react-test-renderer';
import Counter from './main';

import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

describe('<Counter />', () => {
  it('is alive at application start', () => {
    let app = shallow(<Counter />);
    expect(app.find('#buttonup').exists()).toBeTruthy();
  });

  it('increases number on click of increment', () => {
    let app = mount(<Counter />);
    let buttonup = app.find('#buttonup');
    let buttondown = app.find('#buttondown');
    buttonup.simulate('click');
    expect(app.state('count')).toEqual(1);
    buttonup.simulate('click');
    expect(app.state('count')).toEqual(2);

    buttondown.simulate('click');
    expect(app.state('count')).toEqual(1);
    buttondown.simulate('click');
    expect(app.state('count')).toEqual(0);
  });

  it('renders right', () => {
    const tree = renderer.create(<Counter />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
