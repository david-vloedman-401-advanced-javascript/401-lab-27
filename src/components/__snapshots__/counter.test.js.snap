// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> renders right 1`] = `
<div>
  <h3>
    Racecar
  </h3>
  <input
    onChange={[Function]}
  />
  <br />
  <button
    id="buttonup"
    onClick={[Function]}
  >
    Increment
  </button>
  <br />
  <button
    id="buttondown"
    onClick={[Function]}
  >
    Decrement
  </button>
  <h2
    className="negative"
  >
    0
  </h2>
  <input
    onChange={[Function]}
  />
  <br />
  <button
    onClick={[Function]}
  >
    Reset Counter
  </button>
</div>
`;
